test/unittests/t_sync_union_tarball.cc

```diff
@@ -0,0 +1,188 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include <sync_union_tarball.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cassert>
+#include <string>
+
+#include "aux/tar_files.h"
+#include "mock/m_sync_mediator.h"
+#include "sync_item.h"
+#include "util/shared_ptr.h"
+#include "util/string.h"
+
+using ::testing::_;
+using ::testing::DefaultValue;
+using ::testing::Return;
+using ::testing::Property;
+
+using ::publish::SyncItem;
+
+namespace {
+
+class T_SyncUnionTarball : public ::testing::Test {
+ protected:
+  void SetUp() {
+    zlib::Algorithms algo = zlib::kZlibDefault;
+    DefaultValue<zlib::Algorithms>::Set(algo);
+
+    m_sync_mediator_ = new publish::MockSyncMediator();
+  }
+
+  std::string CreateTarFile(const std::string& tar_filename,
+                            const std::string& base64_data) {
+    int tar;
+    std::string data_binary;
+    Debase64(base64_data, &data_binary);
+
+    std::string tmp_dir = CreateTempDir("test_sync_union");
+    assert(!tmp_dir.empty());
```

Comment:
We discuss about using gtest assert. Unfortunately it is not possible since ASSERT_TRUE returns from the function and it does return void, this particular function need to return a std::string